tce/opset/base/base.cc

```diff
@@ -2334,12 +2334,15 @@ END_DEFINE_STATE
 OPERATION_WITH_STATE(HWLOOP, HLOOP_SETUP)
 TRIGGER
     // sanity check
-    if (STATE.hwloopEnabled) {
-        RUNTIME_ERROR("Already hwloop is in progress");
-    }
     if (UINT(1) == 0) {
         RUNTIME_ERROR("hwloop for iterations=0 is invalid");
```

Comment:
-> Hardware loop invocation with 0 iterations is invalid.

Reply:
done